### PR TITLE
[MRG+1] raise explicit TR error

### LINKED ISSUE #1163: Error for unspecfied TR and filtering in signal.clean
may be this error should be more explicit
```Python
from nilearn import datasets, input_data
adhd = datasets.fetch_adhd(n_subjects=1)
masker = input_data.NiftiSpheresMasker(seeds=[(42, 11, 23)], radius=6,
                                       low_pass=.1)
signal = masker.fit_transform(adhd.func[0])
```
gives the following error
>unsupported operand type(s) for /: 'float' and 'NoneType'

### PR DESCRIPTION
**Pull request by salma1601**
fix #1163

### CODE REVIEW

File: nilearn/signal.py [hunk 1]
```diff
@@ -508,6 +508,9 @@ def clean(signals, sessions=None, detrend=True, standardize=True,
                 signals -= confounds.dot(inv).dot(confounds.T).dot(signals)
 
     if low_pass is not None or high_pass is not None:
+        if t_r is None:
+            raise ValueError("Repetition time must be specified for filtering")
```
Comment (GaelVaroquaux) [inline nilearn/signal.py]: Maybe I would say "Repetition time (t_r) must be specified..." to make it more explicit what the user must change.

**Comment by GaelVaroquaux** [comment]
Aside the minor phrasing comment, +1 for merge.

**Comment by GaelVaroquaux** [comment]
+1 for merge

**Comment by KamalakerDadi** [comment]
Can you add simple test please ?

**Comment by KamalakerDadi** [comment]
Does it sounds meaningful to suggest repetition time from image header ?

Some hints I followed from this already existing PR #1036

**Comment by salma1601** [comment]
> Does it sounds meaningful to suggest repetition time from image header ?
> Some hints I followed from this already existing PR #1036

I would rather let the user search himself for this information, to encourage one to know basic information about the data he is working on.
just a personal point of view ;)

**Comment by GaelVaroquaux** [comment]
> I would rather let the user search himself for this information, to
> encourage one to know basic information about the data he is working
> on. just a personal point of view ;)

Somewhat agreed, in particular given that this information is not always
accurate.

**Pull request merged**